app/workers/backend_storage_rewrite_worker.rb

```diff
@@ -1,25 +1,20 @@
+# frozen_string_literal: true
+
 class BackendStorageRewriteWorker
   include Sidekiq::Worker
   sidekiq_options queue: :low
 
-  def self.enqueue_all(providers)
-    batch = Sidekiq::Batch.new
-    batch.description = 'Rewriting Backend Storage'
+  ENQUEUER = ->(class_name, ids) { BackendStorageRewriteWorker.perform_async(class_name, ids) }
 
-    providers.select(:id).find_in_batches do |group|
-      batch.jobs do
-        group.each do |provider|
-          perform_async(provider.id)
-        end
-      end
-    end
+  def self.enqueue_all
+    Backend::StorageRewrite::Processor.new(enqueuer: ENQUEUER).rewrite_all
   end
 
   def self.enqueue(provider_id)
-    perform_async(provider_id)
+    Backend::StorageRewrite::Processor.new(enqueuer: ENQUEUER).rewrite_provider(provider_id)
   end
 
-  def perform(provider_id)
-    Backend::StorageRewrite.rewrite_provider(provider_id)
+  def perform(class_name, ids)
+    Backend::StorageRewrite::Processor.rewrite(class_name, ids)
```

Comment:
This method was receiving a provider before, now it receives a class name and a list of ids, but those ids will belong always to the same provider, that confused me a bit when I read the code first. I think it would be good to add some comments to briefly explain the new design.

Reply:
@jlledom Do you think it would be useful to include the provider ID in the list of worker's arguments? Just for information purposes, because it wouldn't be used.

Reply:
if we don't really care about the provider - better not

Reply:
I prefer not having unused parameters, better just add a comment saying that all `ids` will always belong to the same provider